CRM/Contribute/Page/ContributionPage.php

```diff
@@ -592,17 +592,8 @@ public function whereClause(&$params, $sortBy = TRUE) {
     }
 
     $value = $this->get('financial_type_id');
-    $val = [];
-    if ($value) {
-      if (is_array($value)) {
-        foreach ($value as $k => $v) {
-          if ($v) {
-            $val[$k] = $k;
-          }
-        }
-        $type = implode(',', $val);
-      }
-      // @todo Variable 'type' might not have been defined.
+    if (!empty($value)) {
+      $type = CRM_Utils_Type::validate(implode(',', $value), 'CommaSeparatedIntegers');
```

Comment:
@mlutfy select2 actually returns a comma-separated string to the server, not an array. So I'm not sure how this would work.

Reply:
@colemanw I double-checked, and the `$value` is an array, not CSV.

Could there be magic elsewhere in the form layer?

Reply:
Not sure, but thanks for checking.